controllers/precacheFsm.go

```diff
@@ -60,6 +60,48 @@ func (r *ClusterGroupUpgradeReconciler) precachingFsm(ctx context.Context,
 	clusterGroupUpgrade *ranv1alpha1.ClusterGroupUpgrade) error {
 
 	r.setPrecachingRequired(clusterGroupUpgrade)
+	specCondition := meta.FindStatusCondition(clusterGroupUpgrade.Status.Conditions, "PrecacheSpecValid")
```

Comment:
Could we add this as a constant in the utils package?

Reply:
Done